FaunaDB.Client/Client/RuntimeEnvironmentHeader.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Reflection;
+
+namespace FaunaDB.Client
+{
+    internal class RuntimeEnvironmentHeader
+    {
+        private string runtime;
+        private string driverVersion;
+        private string operatingSystem;
+        private string environment;
+
+        private static RuntimeEnvironmentHeader instance;
+
+        private RuntimeEnvironmentHeader() {}
+
+        private void GatherEnvironmentInfo(IEnvironmentEditor environmentEditor)
+        {
+            this.environment = GetRuntimeEnvironment(environmentEditor);
+            this.operatingSystem = GetOperatingSystemName();
+            this.runtime = GetCurrentRuntime();
+            this.driverVersion = Assembly.Load(new AssemblyName("FaunaDB.Client")).GetName().Version.ToString();
+        }
+
+        private static string GetOperatingSystemName()
+        {
+#if (NETSTANDARD2_1 || NETSTANDARD2_0 || NETSTANDARD1_5)
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return "OSX";
+            }
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return "Linux";
+            }
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return "Windows";
+            }
+            return "Unknown";
+#else
+            // if we're under one of the .net frameworks than OS is Windows
```

Comment:
I don't think I understood this affirmation, what happens if I run on macosx using mono with net45 framework? If I understood the code it will detect as Windows?

Reply:
I was thinking about it as well, but the function `Environment.OSVersion.ToString();` didn't work for me at first.
For some reason, it works now, probably after target frameworks clean-up.
Thanks for pointing this out, I've changed the code so it can detect environments beside Windows